pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/EventService.java

```diff
@@ -0,0 +1,177 @@
+        int ddpInstanceId = ddpInstance.getDdpInstanceIdAsInt();
+        if (!skippedParticipantEventDao.isParticipantEventSkipped(ddpParticipantId, eventName, ddpInstanceId)) {
+            boolean dssSuccessfullyTriggered = triggerDssWithEvent(eventName, ddpInstance, System.currentTimeMillis() / 1000,
+                    ddpParticipantId, ddpParticipantId, null);
+            addParticipantEvent(eventName, ddpInstanceId, ddpParticipantId, dssSuccessfullyTriggered);
+        } else {
+            logger.info("Participant event was skipped for event %s for participant %s ".formatted(eventName, ddpParticipantId)
+                            .concat("based on participant_event table. Dss will not get triggered"));
+            //add this event to the EVENT_QUEUE table, but without triggering dss and flag EVENT_TRIGGERED = false
+            addParticipantEvent(eventName, ddpInstanceId, ddpParticipantId, false);
+        }
+    }
+
+    /**
+     * Sends a POST request to DSS to trigger the event for a kit, using the information in the KitDDPNotification object.
+     * */
+    @VisibleForTesting
+    protected static boolean triggerDssByKitEvent(KitDDPNotification kitDDPNotification) {
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceById(kitDDPNotification.getDdpInstanceId());
+        return triggerDssWithEvent(kitDDPNotification.getEventName(), ddpInstance,
+                kitDDPNotification.getDate() / 1000, kitDDPNotification.getParticipantId(),
+                kitDDPNotification.getDdpKitRequestId(), kitDDPNotification.getUploadReason());
+    }
+
+    /**
+     * Tries to send a POST request to DSS to trigger the event for a kit or a participant,
+     * It will try MAX_TRIES times before giving up and adding the event to the EVENT_QUEUE table.
+     * @param eventType - the type of event to be triggered, e.g. "SALIVA_RECEIVED"
+     * @param eventInfo - the information to be sent in the event, for kit events it is the "ddp_kit_request_id",
+     *      and for participant events it is the ddp_participant_id
+     * @return true if the request was successful, false otherwise
+     * */
+    @VisibleForTesting
+    protected static boolean triggerDssWithEvent(@NonNull String eventType, DDPInstance ddpInstance, long eventDate,
+                                        @NotNull String ddpParticipantId, @NotNull String eventInfo, String reason) {
+        for (int tries = 0; tries < MAX_TRIES; tries++) {
+            try {
+                if (sendDDPEventRequest(eventType, ddpInstance, eventDate, ddpParticipantId, eventInfo, reason)) {
+                    return true;
+                }
+            } catch (Exception e) {
+                logTriggerFailure(ddpInstance, eventType, ddpParticipantId, eventInfo, tries, e);
+            }
+        }
+        logTriggerExhausted(ddpInstance, eventType, ddpParticipantId, eventInfo);
+        return false;
```

Comment:
This method tries to send the post request to DSS for a maximum of MAX_TRIES times, returns true if successful and false if all of the tries fails.